### PR TITLE
[IMP] account: add a credit card account

### PR DESCRIPTION
**Pull request by malb-odoo**
This commit will allow the generation of a credit card journal when either entering manually the information of the card in the iframe or by connecting to a bank.

task: 4197153

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/wizard/setup_wizards.py
Comment (malb-odoo) [inline addons/account/wizard/setup_wizards.py:82]: not sure it's needed :thinking:

File: addons/account/views/res_partner_bank_views.xml
Comment (FlorianGilbert) [inline addons/account/views/res_partner_bank_views.xml:94]: We have to define this function in the account module. Without doing that, users without the `account_online_synchronization` module will have a traceback.

File: addons/account/wizard/setup_wizards.py
Comment (FlorianGilbert) [inline addons/account/wizard/setup_wizards.py:130]: May I  ask you to clean that function and optimize it. 
It seems I miss that `search_count` in the loop :sweat_smile: 
```suggestion
        journals_with_moves = self.env['account.move'].search_fetch(
            [
                ('journal_id', '!=', False),
                ('journal_id.type', '=', journal_type),
            ],
            ['journal_id'],
        ).journal_id
        return self.env['account.journal'].search(
            [
                ('type', '=', journal_type),
                ('bank_account_id', '=', False),
                ('id', 'not in', journals_with_moves.ids)
            ],
            limit=1,
        )
```
I think this change should do the job

File: addons/account/wizard/setup_wizards.py
Comment (FlorianGilbert) [reply addons/account/wizard/setup_wizards.py:82]: I think we should have to move the domain in the bank wizard view.
I mean, we don't use the field in the credit card view (why?) so we don't need domain but if you want to avoid someone using a credit card journal for the bank view, you should move the domain there.

**Review by FlorianGilbert — CHANGES_REQUESTED** [review]

**Review by FlorianGilbert — COMMENTED** [review]
Waiting for security review but lgtm

File: addons/account/views/account_menuitem.xml
Comment (FlorianGilbert) [inline addons/account/views/account_menuitem.xml:47]:
```suggestion
                <menuitem id="menu_action_account_credit_card_journal_form" action="action_new_credit_card_setting" groups="account.group_account_manager" sequence="2"/>
```

File: addons/account/wizard/setup_wizards.py
Comment (FlorianGilbert) [inline addons/account/wizard/setup_wizards.py:142]:
```suggestion
                ('id', 'not in', journals_with_moves.ids),
```

**Review by FlorianGilbert — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**
